### PR TITLE
Add new query option 

### PR DESCRIPTION
**Pull request by xannem**
## Summary
add button to create new query 
Kapture 2023-06-26 at 14 37 19

## How did you test this change?

- [x] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: .changeset/silent-stilts-shush.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-query-builder': patch
+---
+
+Add option to create new query in query header actions
```
Comment (gayathrir11) [inline .changeset/silent-stilts-shush.md:5]: this changeset description would be more relevant to legend-application-query. legend-query-builder doesn't really have the concept of creating new query

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 2]
```diff
@@ -414,6 +419,41 @@ const QueryEditorHeaderContent = observer(
       }
     };
 
+    const newQuery = (): void => {
+      console.log('svpp');
```
Comment (gayathrir11) [inline packages/legend-application-query/src/components/QueryEditor.tsx:423]: can you please remove this

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 3]
```diff
@@ -500,11 +540,21 @@ const QueryEditorHeaderContent = observer(
...
             </div>
           </Button>
-
+          <Button
+            className="query-editor__header__action btn--dark"
+            disabled={editorStore.isPerformingBlockingAction}
+            onClick={cautionNewQuery}
+            title="New query"
+          >
+            <FileEmptyIcon />
```
Comment (gayathrir11) [inline packages/legend-application-query/src/components/QueryEditor.tsx:549]: Not sure about the icon choice here

File: packages/legend-application-query/src/stores/QueryEditorStore.ts [hunk 4]
```diff
@@ -965,6 +966,12 @@ export class ExistingQueryEditorStore extends QueryEditorStore {
     this.query = val;
   }
 
+  setLightName(val: string): void {
```
Comment (gayathrir11) [inline packages/legend-application-query/src/stores/QueryEditorStore.ts:969]: I thinksetlightQueryName would be more more appropriate name for this function

File: .changeset/silent-stilts-shush.md (hunk 1 above)
Comment (xannem) [reply .changeset/silent-stilts-shush.md:5]: True!

File: packages/legend-application-query/src/components/QueryEditor.tsx (hunk 3 above)
Comment (xannem) [reply packages/legend-application-query/src/components/QueryEditor.tsx:549]: me neither, @akphi?

**Review by akphi — CHANGES_REQUESTED** [review]
I think the right thing to do here is to find a way to reset the `QueryEditorStore`. Your current approach is not clean in that it hacks the `ExistingQueryEditorStore`, and the URL is not updated properly.

If we have a way to swap the `ExistingQueryEditorStore` with a `MappingQueryCreatorStore`, and preserve the `GraphManagerState` that would be ideal because we save time by not rebuilding the graph.

But another challenge is that we need to be adaptive here since the query is created from a dataspace, i.e. when we swap out the `QueryEditorStore`, we need to create `DataSpaceQueryCreatorStore` not `MappingQueryCreatorStore`, and we must not re-initialize the graph

I think this is a design-limitation on our end, @MauricioUyaguari and I will see what we should go about here, but thank you so much for attempting it!

**Pull request closed without merging**